app/services/search/postgres/comment.rb

```diff
@@ -74,7 +54,14 @@ def self.search_documents(
         page = page.to_i + 1
         per_page = [(per_page || DEFAULT_PER_PAGE).to_i, MAX_PER_PAGE].min
 
-        relation = ::Comment.joins(ARTICLE_COMMENTABLE_QUERY).where(QUERY_FILTER)
+        relation = ::Comment
+          .where(
+            deleted: false,
+            hidden_by_commentable_user: false,
+            commentable_type: "Article",
+          )
+          .joins("join articles on articles.id = comments.commentable_id")
+          .where("articles.published": true)
```

Comment:
This join is slightly different than before and showed some potential in Blazer and locally to be faster.